### DISCUSSION TITLE
Forward modeling issues

### DISCUSSION THREAD

**Discussion by arthcampos** [slimgroup/JUDI.jl#89] [Q&A] [answered]
Hi,

I've been trying to run the Gauss-Newton inversion example (I know it needs some tweaking to work with newer versions of JUDI) but I have been running into a few issues.  I noticed that the forward modeling operator is mixing the indices in the data vector. Here is an example:
```julia
![example1](https://user-images.githubusercontent.com/17152771/151053041-8d89abaf-303f-4078-ad0f-e9829f572eb3.png)
![example2](https://user-images.githubusercontent.com/17152771/151053049-be5b5e6e-943c-491d-ae45-81bccf4bf1e0.png)

using SegyIO, HDF5, PyPlot, JUDI, LinearAlgebra, Statistics, IterativeSolvers

m, m0, n, d, o = read(h5open("overthrust_model.h5","r"),"m","m0","n","d","o"); 
model0 = Model((n[1],n[2]), (d[1],d[2]), (o[1],o[2]), m0);

block = segy_read("overthrust_shot_records.segy");
d_obs = judiVector(block);

src_geometry = Geometry(block; key="source");
src_data = ricker_wavelet(src_geometry.t[1], src_geometry.dt[1], 0.008f0);
q = judiVector(src_geometry, src_data);

ntComp = get_computational_nt(q.geometry, d_obs.geometry, model0);
info = Info(prod(model0.n), d_obs.nsrc, ntComp);

Pr = judiProjection(info, d_obs.geometry);
Ps = judiProjection(info, q.geometry);
F = judiModeling(info, model0);

d_pred = Pr*F*Ps'*q

figure()
imshow(d_obs.data[8], cmap="seismic", extent=[0,size(d_obs.data[8],2),2,0]); 
ax1 = gca(); ax1[:set_aspect]("auto"); xlabel("Receiver no."); ylabel("Time [s]")
colorbar(); title("Observed data")

figure()
imshow(d_pred.data[8], cmap="seismic", extent=[0,size(d_pred.data[8],2),2,0]); 
ax1 = gca(); ax1[:set_aspect]("auto"); xlabel("Receiver no."); ylabel("Time [s]")
colorbar(); title("Predicted data")
```
The plots show shots in different positions (see attached files). I am doing something wrong here?

**Comment by mloubout** [comment, marked as answer]
What version of JUDI are you using? There was a little ordering issue that was fixed in #77

**Comment by mloubout** [comment]
#87 also fixed some misc issues in the examples (that are now run as  part of CI for safety) so the gauss newton example should run correctly as of version 2.6.4

**Comment by arthcampos** [comment]
Thanks @mloubout! I updated JUDI and the issues were solved. However, I'm running the Gauss-Newton example and the results seem odd with the objective function increasing after a few iterations (contrary to the tutorial from 2018). Is this expected for the example currently?

**Reply by mloubout** [reply]
I'll rerun it to check. Quite a lot has change since 2018 so may need few tweaks. 
Thanks

**Reply by arthcampos** [reply]
@mloubout, I got the same results from the 2018 tutorial changing 
```
p = lsqr!(similar(model0.m), J, d_pred - d_obs; maxiter=maxiter_GN, verbose=true)
```
to
```
p = lsqr!(0 * vec(deepcopy(model0.m.data)), J, d_pred - d_obs; maxiter=maxiter_GN, verbose=true)`
```
in the Gauss-Newton example script. I guess there is probably a better way to do it, but this one did the trick. 

Peace.

**Reply by mloubout** [reply]
Interesting thanks for the update.
